CalendarApp/View Controllers/ScheduleViewController.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  ScheduleViewController.swift
+//  CalendarApp
+//
+//  Created by Angelina Zhang on 7/14/22.
+//
+
+import Foundation
+import CalendarKit
+
+@objc
+public protocol ScheduleSubViewControllerDelegate {
+    func didTapEvent(_ eventID: UUID)
+    func didLongPressEvent(_ eventID: UUID)
+    func fetchEventsForDate(_ date: Date, callback: @escaping(_ events:[CalendarApp.Event]?, _ errorMessage: String?) -> Void)
+}
+
+@objcMembers
+class ScheduleSubViewController : DayViewController {
+    
+    var controllerDelegate: ScheduleSubViewControllerDelegate?
+    private var dateToCalendarKitEvents = [Date: [CalendarKit.Event]]()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    
+    func updateCalendarEvent(_ event: CalendarApp.Event, _ date: Date) {
```

Comment:
Might be easier to work with a type that wraps this logic, similar to `EventDateLogicHandler`.